src/main/java/org/openRealmOfStars/ambient/Bridge.java

```diff
@@ -1020,60 +1030,68 @@ public void updateAllLights() throws IOException {
     String str = new String(buf, StandardCharsets.UTF_8);
     ErrorLogger.debug("Code:" + connection.getResponseCode());
     ErrorLogger.debug(str);
-    Object jsonRoot = parseJsonRoot(buf);
+    Object jsonRoot = parseFirstJsonVal(buf);
     if (!(jsonRoot instanceof JSONObject)) {
       return;
     }
+
     JSONObject json = (JSONObject) jsonRoot;
-    lights = new ArrayList<>();
+    this.lights = new ArrayList<>();
     // Officially Hue bridge supports 50 lights, but let's be on safe side.
     for (int i = 0; i < 255; i++) {
       String lightName = String.valueOf(i);
-      JSONObject member = json.optJSONObject(lightName);
-      if (member != null) {
-        Light light = new Light(lightName);
-        String nameStr = member.optString("name");
-        if (nameStr != null) {
-          light.setHumanReadablename(nameStr);
-        } else {
-          ErrorLogger.debug("No name found for light " + lightName + ".");
-          continue;
-        }
-        JSONObject state = member.optJSONObject("state");
-        if (state != null) {
-          Boolean on = state.optBooleanObject("on");
-          if (on != null) {
-            light.setOn(on.booleanValue());
-          } else {
-            ErrorLogger.debug("No on found for light " + lightName + ".");
-            continue;
-          }
-          Integer bri = state.optIntegerObject("bri", -1);
-          if (bri.intValue() != -1) {
-            light.setBri(bri.intValue());
-          } else {
-            ErrorLogger.debug("No bri found for light " + lightName + ".");
-            continue;
-          }
-          Integer sat = state.optIntegerObject("sat", -1);
-          if (sat.intValue() != -1) {
-            light.setSat(sat.intValue());
-          } else {
-            ErrorLogger.debug("No sat found for light " + lightName + ".");
-            // We only support full color lights
-            continue;
-          }
-          Integer hue = state.optIntegerObject("hue", -1);
-          if (hue.intValue() != -1) {
-            light.setHue(hue.intValue());
-          } else {
-            ErrorLogger.debug("No hue found for light " + lightName + ".");
-            // We only support full color lights
-            continue;
-          }
-        }
-        lights.add(light);
+      JSONObject member = json.optJSONObject(lightName, null);
+      if (member == null) {
+        continue;
+      }
+
+      Light light = new Light(lightName);
+      String nameStr = member.optString("name", null);
+      if (nameStr != null) {
+        light.setHumanReadablename(nameStr);
+      } else {
+        ErrorLogger.debug("No name found for light " + lightName + ".");
+        continue;
       }
+
+      lights.add(light);
```

Comment:
Light should be added only after it has full filled all the required features: on, bri, sat and hue. Last two ones are the most important those are not in all lights.